### ISSUE TITLE
can't sum empty lists

### ISSUE THREAD

**Issue by a-stevan** [nushell/nushell#14463]
### Related problem

i was trying to sum a potentially empty list, which results in the following error
```nushell
[] | math sum
```
```
Error: nu::shell::unsupported_input

  × Unsupported input
   ╭─[entry #10:1:1]
 1 │ [] | math sum
   · ─┬   ────┬───
   ·  │       ╰── Empty input
   ·  ╰── value originates from here
   ╰────
```

i get the issue: because `math sum` can sum lists of integers, durations, etc..., summing an empty list is ambiguous and won't work!

for now, my workaround is 
```nushell
$my_list | if ($in | is-empty) { 0 } else { math sum }
```

> **Note**
>
> the same likely happens with things like `math stddev`, `math median`, ...

### Describe the solution you'd like

i was thinking maybe we could use type information to still be able to sum empty lists?

for instance, if i know my list is a `list<int>`, i think it would be nice if at least the following snippet was working
```nushell
use std assert

let my_list: list<int> = []
assert equal ($my_list | math sum) 0
```
but it gives
```
Error: nu::shell::unsupported_input

  × Unsupported input
   ╭─[entry #1:4:15]
 3 │ let my_list: list<int> = []
 4 │ assert equal ($my_list | math sum) 0
   ·               ────┬───   ────┬───
   ·                   │          ╰── Empty input
   ·                   ╰── value originates from here
   ╰────
```

### Describe alternatives you've considered

see my workaround above

### Additional context and details

the full example would look more like
```nushell
let tab: table<f: string, n: int, a: int, d: int> = ...

let sum = $tab.a | if ($in | is-empty) { 0 } else { math sum }
```
so, because `tab` has its type, we know `$tab.a` is a `list<int>` and thus, `$tab.a | math sum` could work

**Comment by weirdan** [comment]
Another workaround is to use `try`:
```console
$ try { [] | math sum } | default 0
0
```

**Comment by WindSoilder** [comment]
I'm thinking another solution like adding `--start` flag to `math sum`, e.g:
```nushell
> [] | math sum --start 0
0
```
I borrowed the idea from `sum` function in python:
```python
sum(iterable, /, start=0)
    Return the sum of a 'start' value (default: 0) plus an iterable of numbers

    When the iterable is empty, return the start value.
    This function is intended specifically for use with numeric values and may
    reject non-numeric types.
```

**Comment by a-stevan** [comment]
@WindSoilder yeah something like that would also satisfy my definition of "solving this issue" :wink:

**Comment by ClementNerma** [comment]
Hi, is there any news on this? I'm getting errors from a function where I need to sum the numbers in an array (defaulting to 0). Because this happens in a pipeline inside string interpolation, that means I need to extract the expression inside a variable and put a condition around it, which is not very practical.

**Comment by Bahex** [comment]
A very simple solution to this (without requiring a flag like `--start`) is just prepending your starting element. Do instead of this:
```nushell
$my_list | if ($in | is-empty) { 0 } else { math sum }
```
One can do this:
```nushell
$my_list | prepend 0 | math sum
$my_list | prepend 0sec | math sum
# etc
```

I don't think we need to change anything other than adding examples to `math sum` showcasing this.

**Comment by ClementNerma** [comment]
That's a good idea, thanks.
Still, I got bit by this behavior while a program was running, I didn't expect a sum function to throw an error. I believe this should default to zero as that would be a very sensible behavior, or at the very least have a config option to do so (just like we can customize file size units).

**Comment by a-stevan** [comment]
> A very simple solution to this (without requiring a flag like `--start`) is just prepending your starting element
>
that's a very nice way of doing this @Bahex ^_^

**Comment by a-stevan** [comment]
if someone wants to add such an example to the help, i would be happy to close this issue
